src/ee/executors/aggregateexecutor.cpp

```diff
@@ -602,6 +659,36 @@ inline Agg* getAggInstance(Pool& memoryPool, ExpressionType agg_type, bool isDis
         return new (memoryPool) ValuesToCompactAgg();
     case EXPRESSION_TYPE_AGGREGATE_COMPACT_TO_CARDINALITY:
         return new (memoryPool) CompactToCardinalityAgg();
+    case EXPRESSION_TYPE_AGGREGATE_VALUES_TO_TDIGEST:
+        return new (memoryPool) ValuesToTDigestAgg();
+    case EXPRESSION_TYPE_AGGREGATE_MEDIAN:
+        return new (memoryPool) PercentileAgg(50);
+    case EXPRESSION_TYPE_AGGREGATE_TDIGEST_TO_MEDIAN:
+        return new (memoryPool) TDigestToPercentileAgg(50);
```

Comment:
Nice that you found a way to not have to do the actual implementation n times and can parameterize it at least at one level :P

Reply:
Indeed. If VoltDB would allow all of the necessary changes to live a single `.cpp` or `.h` file then I could use the preprocessor to write a template and then stamp it out N times (once for each supported percentile). Alas, the changes are **all over the place** and Java doesn't have a preprocessor.